### PR TITLE
DOC: Don't use Python highlighting for non-python code

### PR DESCRIPTION
**Pull request by eric-wieser**

### CODE REVIEW

File: doc/source/reference/c-api/array.rst [hunk 1]
```diff
@@ -2353,7 +2353,9 @@ it is possible to do this.
 Defining an :c:type:`NpyAuxData` is similar to defining a class in C++,
 but the object semantics have to be tracked manually since the API is in C.
 Here's an example for a function which doubles up an element using
-an element copier function as a primitive.::
+an element copier function as a primitive.
+
+.. code-block:: c
```
Comment (eric-wieser) [inline doc/source/reference/c-api/array.rst:2358]: https://16593-908607-gh.circle-artifacts.com/0/doc/build/html/reference/c-api/array.html#auxiliary-data-with-object-semantics

File: doc/source/reference/c-api/deprecations.rst [hunk 2]
```diff
@@ -48,7 +48,9 @@ warnings).
 
 To use the NPY_NO_DEPRECATED_API mechanism, you need to #define it to
 the target API version of NumPy before #including any NumPy headers.
-If you want to confirm that your code is clean against 1.7, use::
+If you want to confirm that your code is clean against 1.7, use:
+
+.. code-block:: c
 
     #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
```
Comment (eric-wieser) [inline doc/source/reference/c-api/deprecations.rst:55]: https://16593-908607-gh.circle-artifacts.com/0/doc/build/html/reference/c-api/deprecations.html#deprecation-mechanism-npy-no-deprecated-api

File: doc/neps/nep-0000.rst [hunk 3]
```diff
@@ -234,7 +234,9 @@ Header Preamble
 
 Each NEP must begin with a header preamble.  The headers
 must appear in the following order.  Headers marked with ``*`` are
-optional.  All other headers are required. ::
+optional.  All other headers are required.
+
+.. code-block:: rst
```
Comment (eric-wieser) [inline doc/neps/nep-0000.rst:239]: https://16593-908607-gh.circle-artifacts.com/0/doc/neps/_build/html/nep-0000.html#header-preamble

File: doc/neps/nep-0002-warnfix.rst [hunk 4]
```diff
@@ -9,6 +9,8 @@ NEP 2 — A proposal to build numpy without warning with a big set of warning fl
 :Date: 2008-09-04
 :Status: Deferred
 
+.. highlight:: c
```
Comment (eric-wieser) [inline doc/neps/nep-0002-warnfix.rst:12]: https://16593-908607-gh.circle-artifacts.com/0/doc/neps/_build/html/nep-0002-warnfix.html

File: doc/neps/nep-0010-new-iterator-ufunc.rst [hunk 5]
```diff
@@ -393,7 +393,9 @@ The proposed ‘order=’ flags become as follows:
 ‘K’  a layout equivalent to ‘C’ followed by some permutation of the axes, as close to the layout of the input(s) as possible (“Keep Layout”)
 ===  =====================================================================================
 
-Or as an enum::
+Or as an enum:
+
+.. code-block:: c
```
Comment (eric-wieser) [inline doc/neps/nep-0010-new-iterator-ufunc.rst:398]: https://16593-908607-gh.circle-artifacts.com/0/doc/neps/_build/html/nep-0010-new-iterator-ufunc.html#implementation-details

**Comment by charris** [comment]
Thanks Eric.

**Pull request merged**